src/builtins.c

```diff
@@ -416,17 +419,29 @@ static uintptr_t NOINLINE jl_object_id__cold(jl_datatype_t *dt, jl_value_t *v) J
 
 JL_DLLEXPORT inline uintptr_t jl_object_id_(jl_value_t *tv, jl_value_t *v) JL_NOTSAFEPOINT
 {
+    uintptr_t *hash_value = (uintptr_t*)ptrhash_bp(&object_table, v);
+    if(*hash_value != (uintptr_t)HT_NOTFOUND) 
+        return *hash_value;
+    
     jl_datatype_t *dt = (jl_datatype_t*)tv;
-    if (dt == jl_symbol_type)
+    if (dt == jl_symbol_type) {
+        ptrhash_put(&object_table, v, (void *)((jl_sym_t*)v)->hash);
```

Comment:
There is no point in putting simple things in the cache, only recursive ones with interior pointers.
```suggestion
```